dropshot/examples/https.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020 Oxide Computer Company
+
+/*!
+ * Example use of Dropshot with TLS enabled
+ */
+
+use dropshot::endpoint;
+use dropshot::ApiDescription;
+use dropshot::ConfigDropshot;
+use dropshot::ConfigLogging;
+use dropshot::ConfigLoggingLevel;
+use dropshot::HttpError;
+use dropshot::HttpResponseOk;
+use dropshot::HttpResponseUpdatedNoContent;
+use dropshot::HttpServerStarter;
+use dropshot::RequestContext;
+use dropshot::TypedBody;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use serde::Serialize;
+use std::io::Write;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+use tempfile::NamedTempFile;
+
+/*
+ * This function would not be used in a normal application. It is used to
+ * generate temporary keys and certificates for the purpose of this demo.
+ */
+fn generate_keys() -> Result<(NamedTempFile, NamedTempFile), String> {
+    let keypair =
+        rcgen::generate_simple_self_signed(vec!["localhost".to_string()])
+            .map_err(|e| e.to_string())?;
+    let cert = keypair.serialize_pem().map_err(|e| e.to_string())?;
+    let priv_key = keypair.serialize_private_key_pem();
+
+    let mut cert_file =
+        NamedTempFile::new().map_err(|_| "failed to create cert_file")?;
```

Comment:
I really wish there were a way to include a name with this file.  That way if this crashes and someone finds these files they'll have some idea what they are and that they can probably delete them.  But I don't see an easy way to do this with the `tempfile` crate.  Oh well.

Reply:
I found one, using tempfile::Builder. I'll make use of it here and in the tests